src/cdk-experimental/menu/menu-item.ts

```diff
@@ -50,9 +53,52 @@ export class CdkMenuItem {
   }
   private _checked = false;
 
+  /**  Whether the CdkMenuItem is disabled - defaults to false */
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+  private _disabled = false;
+
   /** Emits when the attached submenu is opened */
   @Output() opened: EventEmitter<void> = new EventEmitter();
 
+  constructor(
+    /** reference a parent CdkMenuGroup component */
+    private readonly _menuGroup: CdkMenuGroup
+  ) {}
+
+  /** Configure event subscriptions */
+  ngAfterContentInit() {
+    if (this.role !== 'menuitem') {
+      this._menuGroup.change.subscribe((button: CdkMenuItem) => this._toggleCheckedState(button));
+    }
+  }
+
+  /**
+   * If the role is menuitemcheckbox or menuitemradio and not disabled, emits a change event
+   * on the enclosing parent MenuGroup.
+   */
+  trigger() {
+    if (this.disabled) {
+      return;
+    }
+
+    if (this.hasSubmenu()) {
+      // TODO(andy): open the menu
+    } else if (this.role !== 'menuitem') {
+      this._menuGroup.change.next(this);
```

Comment:
This is quite a clever mechanism -- I like it. (I wouldn't have thought of emitting menu item instances through an event emitter that way.)

That said, it feels a little awkward to reach into the CdkMenuGroup and call a method directly on one of its members (even if it's a public member). I might instead have a 'registerTriggeredMenuItem()' method on the menu group class -- calling it with a menu item causes the item to be emitted by (change).

Then again, since this is all package-private interaction, it may be okay this way. I'll leave it to your judgment.

Reply:
Good Idea. I took it a step further and moved the logic into CdkMenuGroup - I think it makes more sense for it to determine when to emit a change event than having the MenuItem doing so.